cypress/integration/listview/downloadModal.spec.js

```diff
@@ -0,0 +1,50 @@
+import path from 'path';
+
+const EDITABLE_URL =
+    '/iframe.html?id=modules-listview-stories-downloadmodal--default-download-modal&viewMode=story';
+
+describe('DownloadModal', () => {
+    beforeEach(() => {
+        cy.visit(EDITABLE_URL);
+    });
+
+    it('should open DownloadModal and download a data.json file', () => {
+        const downloadsFolder = Cypress.config('downloadsFolder');
+        cy.get('[id="download-button-icon"]').click();
+        cy.get('h1').should('contain', 'Download');
+        cy.get('p').should(
+            'contain',
+            'The entries matching will be downloaded. If you need over 10,000 entries consider exporting.',
+        );
+        cy.get('h2').should('have.length', 2);
+        cy.get('h2').eq(0).should('contain', 'Maximum entries');
+        cy.get('h2').eq(1).should('contain', 'Select format');
+        cy.get('[type="number"]').should('have.value', '500');
+        cy.get('[type="radio"]').eq(0).should('be.checked').and('have.value', 'json');
+        cy.get('[type="radio"]').eq(1).should('not.be.checked').and('have.value', 'csv');
+        cy.get('button').contains('Download').click();
```

Comment:
we need to scope these selectors maybe something like:
cy.get('#modal-id').within(() => {
// then here use all these selectors
})
for this, you need to pass a new prop id that is passed to the internal Modal